src/docfx/config/ops/OpsAccessor.cs

```diff
@@ -225,43 +217,29 @@ async Task<HttpResponseMessage> SendRequest(HttpRequestMessage request)
             }
         }
 
-        private static HttpMiddleware BuildMiddleware(DocsEnvironment? environment = null, HttpMiddleware? middleware = null)
+        private static HttpMiddleware BuildMiddleware(HttpMiddleware? middleware = null)
         {
             return async (request, next) =>
             {
                 // Default header which allows fallback to public data when credential is not provided.
                 request.Headers.TryAddWithoutValidation("X-OP-FallbackToPublicData", "True");
-
-                // don't access key vault for osx since azure-cli will crash in osx
-                // https://github.com/Azure/azure-cli/issues/7519
-                if (!RuntimeInformation.IsOSPlatform(OSPlatform.OSX) &&
-                    !request.Headers.Contains("X-OP-BuildUserToken"))
+                if (!request.Headers.Contains("X-OP-BuildUserToken"))
                 {
-                    // For development usage
-                    try
-                    {
-                        environment ??= DocsEnvironment;
-                        var token = environment switch
-                        {
-                            DocsEnvironment.Prod => s_opsTokenPublic,
-                            DocsEnvironment.PPE => s_opsTokenPubDev,
-                            DocsEnvironment.Perf => s_opsTokenPerf,
-                            _ => throw new InvalidOperationException(),
-                        };
-
-                        request.Headers.Add("X-OP-BuildUserToken", await token.Value);
-                    }
-                    catch (Exception ex)
-                    {
-                        Log.Write($"Cannot get 'OPBuildUserToken' from azure key vault, please make sure you have been granted the permission to access.");
-                        Log.Write(ex);
-                    }
+                    request.Headers.TryAddWithoutValidation("Authorization", $"Bearer {await GetAccessTokenAsync()}");
                 }
 
                 return await (middleware != null ? middleware(request, next) : next(request));
             };
         }
 
+        private static async Task<string> GetAccessTokenAsync()
```

Comment:
Ensure this works for users using vscode extension who does not login. I don't remember exactly in that case did vscode extension not pass `X-OP-BuildUserToken` or it passes a well known public user `X-OP-BuildUserToken`. cc @928PJY

Reply:
Good point, for the public user, We just don't pass the token. We should not throw exceptions if we cannot get the AAD token